extension/content/components/recipes/RecipeEditor.js

```diff
@@ -0,0 +1,189 @@
+import React, { useState, useEffect } from "react";
+import PropTypes from "prop-types";
+import {
+  Alert,
+  Button,
+  ButtonToolbar,
+  Form,
+  FormGroup,
+  FormControl,
+  ControlLabel,
+  HelpBlock,
+  InputPicker,
+} from "rsuite";
+import {
+  UnControlled as CodeMirrorUC,
+  Controlled as CodeMirror,
+} from "react-codemirror2";
+import { useSelectedEnvironmentAPI } from "devtools/contexts/environment";
+
+export default function RecipeEditor(props) {
+  const { match } = props;
+  const api = useSelectedEnvironmentAPI();
+  const [data, setData] = useState({ arguments: {} });
+  const [actions, setActions] = useState([]);
+
+  async function getActionsOptions() {
+    const res = await api.fetchActions();
+    const actions = res.results.map(action => ({
+      label: action.name,
+      value: action.id,
+    }));
+
+    setActions(actions);
+  }
+
+  async function getRecipe(id) {
+    const recipe = await api.fetchRecipe(id);
+    setData(recipe.latest_revision);
+  }
+
+  useEffect(() => {
+    getActionsOptions();
+    if (match.params.id) {
+      getRecipe(match.params.id);
+    }
+  }, []);
+
+  const handleActionIDChange = value => {
+    setData({
+      ...data,
+      action: {
+        ...data.action,
+        id: value,
+      },
+    });
+  };
+  const handleChange = (key, value) => {
+    setData({ ...data, [key]: value });
+  };
+
+  const handleSubmit = () => {
+    const id = match.params.id;
+    const requestBody = formatRequestBody();
+    const requestSave = api.saveRecipe(id, requestBody);
+
+    requestSave
+      .then(() => {
+        location.replace("/content.html#");
+        Alert.success("Changes Saved");
+      })
+      .catch(err => {
+        Alert.error(`An Error Occurred: ${JSON.stringify(err.data)}`, 5000);
+      });
+  };
+
+  const formatRequestBody = () => {
+    /* eslint-disable no-unused-vars */
+    const { ["comment"]: _omitData, ...requestBody } = data;
```

Comment:
this should be:
```
const { comment: _omitComment, ...cleanedData } = data;
```

primarily the change to remove the square braces but I think those names are also a little clearer

also maybe include a TODO about removing this once https://github.com/mozilla/normandy/issues/2189 is addressed

Reply:
After talking to @mythmon we've decided we should keep the comment field so I'll file an issue for how to deal with it. no TODO required.